core/amber/src/main/python/core/python_worker.py

```diff
@@ -1,11 +1,32 @@
 from overrides import overrides
-from threading import Thread
+from threading import Thread, Event
+from loguru import logger
 
 from core.models.internal_queue import InternalQueue
 from core.runnables import MainLoop, NetworkReceiver, NetworkSender
 from core.util.runnable.runnable import Runnable
 from core.util.stoppable.stoppable import Stoppable
 
+import os
+import sys
+import signal
+import psutil
+import asyncio
+
+
+def self_clean_child_process(code: int):
+    current_process = psutil.Process()
+    children = current_process.children(recursive=True)
+    for child in children:
+        if child.is_running():
+            try:
+                os.kill(child.pid, signal.SIGKILL)
+            except Exception as e:
+                logger.info(
+                    f"Exception during process termination PID {str(child.pid)}: {e} "
+                )
+    sys.exit(code)
```

Comment:
Minor comment. I see you pass 0 as the exit code for all situations, which means it exits without error. It would make more sense to pass a non-0 exit code when exited abnormally.